### PR TITLE
EIP-1898: Allow default block parameter to be blockHash

### PR DESCRIPTION
**Pull request by terorie**
"Allow default block parameter to be blockHash"

https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1898.md

Backports https://github.com/openethereum/openethereum/pull/10932

### CODE REVIEW

**Comment by terorie** [comment]
I hope this is the right place for proposing changes to 3.1 beta. Please let me know if I should file it against the main repo instead.

**Comment by adria0** [comment]
> I hope this is the right place for proposing changes to 3.1 beta. Please let me know if I should file it against the main repo instead.

Hi @terorie, the `3.1.0` branch was frozen in September 30th https://github.com/openethereum/openethereum/issues/11881#issue-711820024 , so this will go to into `3.2.0`

File: rpc/src/v1/impls/eth.rs [hunk 1]
```diff
@@ -540,6 +542,23 @@ where
...
+            hash,
+            require_canonical,
+        } => {
+            // block check takes precedence over canon check.
+            match client.block_status(BlockId::Hash(hash.clone())) {
+                BlockStatus::InChain => {}
+                _ => return Err(errors::unknown_block()),
+            };
+
+            // TODO Check block hash canonical
```
Comment (adria0) [inline rpc/src/v1/impls/eth.rs:555]: Should be this uncommented?

https://github.com/openethereum/openethereum/pull/10932/files#diff-efc3317730d3b6b64e36557f32d51ffb36c70bf146048c15790d4827fa1f5e12R486-R488

**Comment by rakita** [comment]
Backported it here with canonical get: https://github.com/openethereum/openethereum/pull/203

**Pull request closed without merging**